x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/WaitForIndexGreenStep.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.ilm;
+
+import com.carrotsearch.hppc.cursors.ObjectCursor;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.routing.IndexRoutingTable;
+import org.elasticsearch.cluster.routing.IndexShardRoutingTable;
+import org.elasticsearch.cluster.routing.RoutingTable;
+import org.elasticsearch.cluster.routing.ShardRouting;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.Index;
+
+import java.io.IOException;
+import java.util.Objects;
+
+class WaitForIndexGreenStep extends ClusterStateWaitStep {
```

Comment:
I think rather than having a separate `WaitForIndexGreenStep` we can transform the existing `WaitForIndexYellowStep` into a `WaitForIndexColorStep` that takes an enum parameter for color (GREEN, RED, YELLOW). Otherwise, this is essentially a duplicate of that class.

Reply:
Totally agree. Rewritten that.